### ISSUE TITLE
Invalid URLs in kubeflow/website

### ISSUE THREAD

**Issue by zhenghuiwang** [kubeflow/website#1683]
I wrote a [script](https://github.com/kubeflow/website/pull/1684) to scan all `.md` files in this repository, extract `http/https` URLs, and ping them for validation.

Out of 1300+ URLs, here is a list of URLs that seem to be invalid. **Warning: false negative is possible.** All file path is under
`kubeflow/website/content/docs`

**Feel free to remove any URL from this list, if it is found valid.**

| File Path        | URL           | Request Status  |
| ---------------------------------------------------------- |:---------------------------------------------------------------------| -----------:|
|/pipelines/sdk/output-viewer.md | https://github.com/kubeflow/pipelines/blob/master/components/dataflow/tfma/src/model_analysis.py | 404|
|/pipelines/sdk/output-viewer.md | https://github.com/kubeflow/pipelines/blob/master/components/dataflow/predict/src/predict.py | 404|
|/pipelines/sdk/component-development.md | https://github.com/kubeflow/pipelines/blob/master/samples/tutorials/Data | 404|
|/pipelines/sdk/sdk-overview.md | https://github.com/kubeflow/pipelines/blob/master/samples/core/component_build/component_build.ipynb | 404|
|/pipelines/sdk/sdk-overview.md | https://github.com/kubeflow/pipelines/blob/master/samples/core/xgboost-spark/xgboost-training-cm.py | 404|
|/pipelines/tutorials/build-pipeline.md | https://github.com/kubeflow/pipelines/tree/master/samples/core. | 404|
|/pipelines/tutorials/build-pipeline.md | https://github.com/kubeflow/pipelines/blob/master/samples/core/tfx-oss/TFX | 404|
|/about/community.md | https://calendar.google.com/calendar/embed?src | 400|
|/about/community.md | https://calendar.google.com/calendar/ical/kubeflow.org_7l5vnbn8suj2se10sen81d9428 | 400|
|/about/contributing.md | https://cla.developers.google.com/ | 307|
|/about/contributing.md | https://k8s-gubernator.appspot.com/pr | 405|
|/about/events.md | https://drive.google.com/open?id | 404|
|/about/events.md | https://schd.ws/hosted_files/kccncchina2018english/0a/Kubecon | 404|
|/about/events.md | https://schd.ws/hosted_files/kccncchina2018english/fe/Kubecon | 404|
|/about/events.md | https://kafka-summit.org/ | 403|
|/about/events.md | https://scipy2018.scipy.org/ehome/index.php?eventid | 403|
|/about/events.md | https://schd.ws/hosted_files/kccnceu18/c2/Building | 404|
|/about/events.md | https://schd.ws/hosted_files/kccnceu18/a1/KubeCon_EU_2018 | 404|
|/ibm/install-kubeflow.md | https://console.bluemix.net/docs/cli/index.html#overview | 308|
|/ibm/install-kubeflow.md | https://console.bluemix.net/docs/containers/cs_regions.html#regions-and-zones | 308|
|/ibm/install-kubeflow.md | https://console.bluemix.net/docs/containers/cs_regions.html#zones | 308|
|/ibm/install-kubeflow.md | https://github.com/kubeflow/kubeflow/releases/tag/ | 404|
|/components/misc/nuclio.md | https://github.com/v3io/tutorials/blob/master/demos/stocks/read-tweets.ipynb | 404|
|/components/misc/nuclio.md | https://www.iguazio.com/ | 403|
|/components/misc/nuclio.md | http://nuclio-dashboard.nuclio.svc | FAIL|
|/components/misc/nuclio.md | https://github.com/v3io/tutorials/blob/master/demos/image-classification/02-infer.ipynb | 404|
|/components/misc/nuclio.md | https://github.com/v3io/tutorials/blob/master/demos/netops/04-infer.ipynb | 404|
|/components/hyperparameter-tuning/experiment.md | https://github.com/kubeflow/katib/blob/master/examples/v1alpha3/nasjob-example-RL.yaml | 404|
|/components/serving/kfserving.md | https://raw.githubusercontent.com/kubeflow/kfserving/master/install/$TAG/kfserving.yaml | 404|
|/components/serving/overview.md | https://www.seldon.io/tech/products/deploy/ | 403|
|/components/serving/tfserving_new.md | http://EXTERNAL_IP | FAIL|
|/components/serving/tfserving_new.md | https://YOUR_HOST/tfserving/models/mnist | FAIL|
|/components/central-dash/overview.md | https://example.mydomain.com | FAIL|
|/components/central-dash/overview.md | https://example.mydomain.com/myapp | FAIL|
|/gke/gcp-e2e.md | https://github.com/kubeflow/kubeflow/tree/master/kubeflow/core | 404|
|/gke/gcp-e2e.md | https://github.com/kubeflow/kubeflow/blob/master/kubeflow/examples/prototypes/tf-job-simple.jsonnet | 404|
|/gke/gcp-e2e.md | https://github.com/kubeflow/kubeflow/tree/master/kubeflow/tf-serving | 404|
|/gke/gcp-e2e.md | https://github.com/kubeflow/kubeflow/blob/master/scripts/gke/deploy.sh | 404|
|/gke/customizing-gke.md | https://github.com/kubeflow/manifests/tree/ | 404|
|/gke/customizing-gke.md | https://github.com/kubeflow/manifests/tree/ | 404|
|/gke/customizing-gke.md | https://github.com/kubeflow/manifests/tree/ | 404|
|/gke/troubleshooting-gke.md | https://github.com/kubernetes/contrib/blob/ | 404|
|/gke/troubleshooting-gke.md | http://envoy | FAIL|
|/gke/troubleshooting-gke.md | https://file.googleapis.com/v1beta1/projects/cloud-ml-dev/locations/us-central1-a/instances | 404|
|/gke/troubleshooting-gke.md | https://github.com/kubeflow/kubeflow/blob/master/scripts/gke/deployment_manager_configs/cluster-kubeflow.yaml#L31 | 404|
|/gke/troubleshooting-gke.md | https://github.com/kubeflow/kubeflow/blob/master/scripts/gke/deployment_manager_configs/cluster.jinja#L105 | 404|
|/gke/private-clusters.md | https://www.googleapis.com/compute/v1/projects/$ | 401|
|/gke/monitoring.md | https://github.com/kubeflow/kubeflow/blob/master/kubeflow/gcp/prometheus.libsonnet | 404|
|/gke/monitoring.md | https://github.com/kubeflow/kubeflow/blob/master/kubeflow/gcp/prometheus.yml#L75 | 404|
|/gke/monitoring.md | https://github.com/kubeflow/kubeflow/blob/master/kubeflow/gcp/metric-collector.libsonnet#L83 | 404|
|/gke/monitoring.md | https://github.com/kubeflow/kubeflow/blob/master/kubeflow/gcp/prometheus.yml#L127 | 404|
|/gke/monitoring.md | https://github.com/kubeflow/kubeflow/blob/master/kubeflow/gcp/prometheus.libsonnet#L170 | 404|
|/gke/deploy/oauth-setup.md | https://iap.googleapis.com/v1/oauth/clientIds/ | 404|
|/gke/deploy/monitor-iap-setup.md | https://mykubeflow.endpoints.myproject.cloud.goog/_gcp_gatekeeper/authenticate | FAIL|
|/examples/blog-posts.md | https://www.iguazio.com/blog/orchestrating-ml-pipelines-scale-kubeflow/ | 403|
|/examples/blog-posts.md | https://cloud.google.com/blog/products/ai-machine-learning/getting-started-kubeflow-pipelines | 405|
|/other-guides/troubleshooting.md | https://github.com/kubeflow/kubeflow/blob/master/scripts/gke/deployment_manager_configs/cluster.jinja#L105 | 404|
|/other-guides/support.md | https://www.seldon.io/ | 403|
|/other-guides/job-scheduling.md | https://github.com/volcano-sh/volcano/blob/master/pkg/apis/scheduling/v1alpha2/types.go | 404|
|/other-guides/kustomize.md | https://github.com/kubeflow/manifests/blob/master/kfdef/kfctl_existing_arrikto.yaml | 404|

/cc @sarahmaddox @abhi-g

**Comment by abhi-g** [comment]
Thanks Zhenghui, this is quite useful. I don't think there is an automated
way to update the links, but periodically running this script can be super
informative to track hygiene. Hopefully we can have the links fixed up even
if it is a manual effort. Could this issue be a good candidate for "help
wanted"?

**Comment by sarahmaddox** [comment]
Thanks @zhenghuiwang This is very useful indeed. 

@abhi-g Good idea: I've labeled this issue as "help wanted".

**Comment by janeman98** [comment]
The first two broken links are reported in issue #1687 as well.

@sarahmaddox , I can help on this.

**Comment by sarahmaddox** [comment]
Thanks @janeman98 
/assign @janeman98

**Comment by Jeffwan** [comment]
> /aws/authentication.md | https://www.shanjiaxin.com | FAIL

It's plain text, just a custom domain example. This can be skipped. 

> /aws/logging.md | https://console.aws.amazon.com/cloudwatch/home#logs

This link need user to login AWS console and can be skipped as well.

**Comment by zhenghuiwang** [comment]
> > /aws/authentication.md | https://www.shanjiaxin.com | FAIL
> 
> It's plain text, just a custom domain example. This can be skipped.
> 
> > /aws/logging.md | https://console.aws.amazon.com/cloudwatch/home#logs
> 
> This link need user to login AWS console and can be skipped as well.

Removed from the list. Thanks

**Comment by janeman98** [comment]
@zhenghuiwang , @sarahmaddox 

I have visited and fixed all the broken urls. The only open issue is [1687](https://github.com/kubeflow/website/issues/1687) . Can we close this issue?

**Comment by sarahmaddox** [comment]
Thanks @janeman98!
/close

**Comment by janeman98** [comment]
Thanks @sarahmaddox !
